Add a basic integration test to validate style-dictionary CSS output

hannajones opened the pull request:
This patch adds a basic Jest test that can be used to validate the output of running `style-dictionary build` with our custom configuration. It's kind of an integration style test since we're actually checking a built CSS file against an expected fixture value. This should give us an easy way to do TDD, and should also give us confidence that any transforms/formatters we introduce aren't producing unintended side effects.

To run the test locally you can just run `npm test`. If you want to see how things look when the test fails, either modify a value in `design-tokens.json`, or modify a few lines of the fixture in `config.test.js`. You should see an error message that points to the lines that are different:

<img width="697" alt="Screenshot 2024-01-24 at 1 56 31 PM" src="https://github.com/FirefoxUX/gecko-dev/assets/15138046/943ec242-54e8-4220-b798-b688189a056a">

I'm using an inline snapshot for convenience. If you're making a bunch of changes to the JSON file and you're confident that the diff between the built CSS and the fixture is desirable you can update the fixture automatically by running `npm test -- -u`.

hannajones committed:
Add a basic integration test to validate style-dictionary CSS output

TGiles reviewed (approved):
This looks like a good starting point for adding more tests, thanks!

TGiles reviewed (approved):
(whoops, double approval thanks to the vscode github extension 😂)

mstriemer reviewed (approved):
This is awesome, thanks for adding this!

I see different ordering from what's committed when I run `npm run build` unfortunately, I wonder if it varies by platform somehow? Hopefully running the output through a formatter that alphabetizes the lines will fix that

hannajones commented:
> I see different ordering from what's committed when I run `npm run build` unfortunately, I wonder if it varies by platform somehow? Hopefully running the output through a formatter that alphabetizes the lines will fix that

@mstriemer I was a little worried about that since `style-dictionary` doesn't guarantee order...I think it will be easier to sort out once we have a formatter in place. Worst case down the line I can figure out a way to just check that all the `--variable-name: value` pairs have the expected value, but it would be nice to be able to test the exact contents of the built CSS file.

hannajones commented:
After rebasing I changed the test to check values in an object and removed the snapshot - this seems more useful for now while we don't care about order. This is an example of the kind of output I get if I remove @TGiles's light-dark transform - seems pretty useful imo.

@mstriemer can you confirm this works on Windows/whatever platform you were testing yesterday? 

<img width="710" alt="Screenshot 2024-01-25 at 10 41 37 AM" src="https://github.com/FirefoxUX/gecko-dev/assets/15138046/94210bbd-4a8b-49ce-bde3-f7bc2981a93c">
